### PR TITLE
[#4367] rodsLog cleanup (main)

### PR DESCRIPTION
**Pull request by epsilon-phase**
I haven't gotten everything, but at least there'll be a few less :)

### CODE REVIEW

**Comment by korydraughn** [comment]
Ignore files such as `lsUtil.cpp`. Those are specific to the icommands and should not contain references to the logging API (because the logging API is not designed for client-side use).

File: server/api/src/rsGeneralAdmin.cpp [hunk 1]
```diff
@@ -428,7 +428,9 @@ int _addChildToResource(generalAdminInp_t* _generalAdminInp, rsComm_t* _rsComm)
 
     resc_input[irods::RESOURCE_CHILDREN] = rescChildren;
 
-    log_api::info("rsGeneralAdmin add child \"{}\" to resource \"{}\"", _generalAdminInp->arg3, _generalAdminInp->arg2);
+    log_api::info(R"stuff(rsGeneralAdmin add child "{}" to resource "{}")stuff",
```
Comment (trel) [inline server/api/src/rsGeneralAdmin.cpp:431]: stuff?

**Comment by epsilon-phase** [comment]
Raw string literals are funny huh? Does "delimiter()delimiter" work better?

-------- Original Message --------

**Comment by korydraughn** [comment]
The delimiters are optional. If you prefer to use them, perhaps using 2 or more underscores is a good option since underscores tend to be less noisy.

File: lib/core/include/irods/irods_logger.hpp [hunk 2]
```diff
@@ -43,6 +43,7 @@ namespace irods::experimental
 
         enum class level
         {
+            sql,
```
Comment (korydraughn) [inline lib/core/include/irods/irods_logger.hpp:46]: What is the motivation behind these changes to the logging API?

Comment (epsilon-phase) [reply lib/core/include/irods/irods_logger.hpp:46]: The sql loglevel is distinct in the old log api, I would guess because it's even more prone to filling the logs than trace.

Comment (korydraughn) [reply lib/core/include/irods/irods_logger.hpp:46]: The log levels are pretty standard, so we don't want to introduce one people are unfamiliar with.

A better approach would be to introduce a new category.

**Comment by trel** [comment]
this is looking pretty good.   the main use case for the LOG_SQL thing was to see the bind variables and sequel strings being sent to the database - that should probably be debug rather than trace...  if you haven't already touched that area.

File: plugins/api/src/register_physical_path.cpp [hunk 3]
```diff
@@ -21,56 +21,59 @@
...
-#define IRODS_REPLICA_ENABLE_SERVER_SIDE_API
-#include "irods/replica_proxy.hpp"
-
-#include <boost/lexical_cast.hpp>
-#include <fmt/format.h>
-
-#include <iostream>
+#  include "irods/irods_logger.hpp"
+#  include "irods/irods_resource_backport.hpp"
+#  include "irods/irods_resource_redirect.hpp"
```
Comment (korydraughn) [inline plugins/api/src/register_physical_path.cpp:26]: Let's revert the formatting on all these include directives.

Right now, they look weird in relation to the include directives that weren't touched.

File: server/api/src/rsDataObjRename.cpp [hunk 4]
```diff
@@ -47,8 +47,6 @@
 #include <cstring>
 #include <chrono>
 
-using logger = irods::experimental::log;
```
Comment (korydraughn) [inline server/api/src/rsDataObjRename.cpp:50]: Shouldn't this be `using log_api = irods::experimental::log::api;` instead of being removed?

File: server/api/src/rsGetRescQuota.cpp [hunk 5]
```diff
@@ -12,6 +13,8 @@
 
 #include <cstring>
 
+using log_server = irods::experimental::log::server;
```
Comment (korydraughn) [inline server/api/src/rsGetRescQuota.cpp:16]: Why the `server` category instead of the `api` category?

**Review by korydraughn — CHANGES_REQUESTED** [review]
Excellent! This is going to make dealing with the log so much better.

File: server/api/src/rsDataObjRename.cpp (hunk 4 above)
Comment (epsilon-phase) [reply server/api/src/rsDataObjRename.cpp:50]: There was a name clash that developed for some reason. I'll change the style over :)

File: server/api/src/rsDataObjRename.cpp [hunk 6]
```diff
@@ -47,7 +47,7 @@
...
+using log_api = log_api;
```
Comment (korydraughn) [inline server/api/src/rsDataObjRename.cpp:50]: I see now. Some other header is defining this alias, right?

Let's explicitly define the alias so there's no dependency on a header providing it.
```cpp
using log_api = irods::experimental::log::api;
```

**Comment by korydraughn** [comment]
If the changes compile for each OS (under the github actions) then squash the commits and leave off the pound.

Have you verified that the core tests pass with these changes?

**Review by alanking — COMMENTED** [review]
Looks real good. Agreed with @korydraughn's comment regarding tests. If this is building and passing tests then I think it's pretty much there. Any thoughts from others on keeping the addition of the new sql category in its own commit separate from the `rodsLog` replacements?

Along those lines, we should probably make an issue to update our documentation with respect to `spLogSql` since it will no longer be the way to enable SQL logging. Also, the new category should be represented in the docs anyway.

**Comment by korydraughn** [comment]
> Any thoughts from others on keeping the addition of the new sql category in its own commit separate from the rodsLog replacements?

That's not a bad idea. Let's do that.

> Along those lines, we should probably make an issue to update our documentation with respect to `spLogSql` since it will no longer be the way to enable SQL logging. Also, the new category should be represented in the docs anyway.

Yes, we need an issue to update the docs, but I'm not sure this PR obsoletes `spLogSql` just yet. I didn't see anything in there that removed the need for that option. I only saw changes regarding the operations (e.g. `log_sql::debug("chlSomeOperation() 1")`). Did you spot the code that removes the need for that option? If so, I'm sure some tests are going to fail.

**Comment by trel** [comment]
Two eventual commits seems good (one for new sql category, one for changing all the things).

**Comment by alanking** [comment]
> Did you spot the code that removes the need for that option?

We added a new category for SQL logging and then replaced many calls, but it turns out that we haven't replaced all of the calls in this PR (and that's okay). Once all calls to `rodsLog(LOG_SQL, ...)` and `rodsLogSql` are replaced, the option will not be useful anymore. It only exists to allow for `LOG_SQL` to have an effect in `rodsLog` because `LOG_SQL` is level 11 (the highest available log level). Was just mentioning that we should probably update the documentation once this has happened, and maybe that's just part of #4367.

**Comment by korydraughn** [comment]
Let's capture the documentation effort for `spLogSql` in a new issue.

@epsilon-phase Please use the issue number 4367 for the sql category commit as well.

**Review by alanking — APPROVED** [review]
